modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaClientCodegen.java

```diff
@@ -152,13 +152,15 @@ public void processOpts() {
         writeOptional(outputFolder, new SupportingFile("manifest.mustache", projectFolder, "AndroidManifest.xml"));
         supportingFiles.add(new SupportingFile("travis.mustache", "", ".travis.yml"));
         supportingFiles.add(new SupportingFile("ApiClient.mustache", invokerFolder, "ApiClient.java"));
+        supportingFiles.add(new SupportingFile("ApiClientHolder.mustache", invokerFolder, "ApiClientHolder.java"));
         if(!"resttemplate".equals(getLibrary())) {
             supportingFiles.add(new SupportingFile("StringUtil.mustache", invokerFolder, "StringUtil.java"));
         }
 
         supportingFiles.add(new SupportingFile("auth/HttpBasicAuth.mustache", authFolder, "HttpBasicAuth.java"));
         supportingFiles.add(new SupportingFile("auth/ApiKeyAuth.mustache", authFolder, "ApiKeyAuth.java"));
         supportingFiles.add(new SupportingFile("auth/OAuth.mustache", authFolder, "OAuth.java"));
+        supportingFiles.add(new SupportingFile("auth/OAuthBearerProvider.mustache", authFolder, "OAuthBearerProvider.java"));
```

Comment:
This file shouldn't be added for retrofit library

Reply:
I can't make this conditional at the moment as OAuthBearerProvider is required to meaningfully implement OAuth and thus OAuth depends on this interface now - I don't seem to have identified a way to sensibly attach myself within OAuth lifecycle to supply new Bearer - one at the first server hit, a renewed one once the previous one expires.

As this is an interface only and very crucial for our needs, can you please elaborate more on why this can't be included when retrofit library is in place? Maybe I'm overlooking some basic concept..

Reply:
Retrofit has its own OAuth class that doesn't depend on OAuthBearerProvider so this would be dead code.

Reply:
Oh, ok. That means it does not have even OAuth itself, right? So the same rules should be applicable as for OAuth class, which is in there already. Or am I missing something?

Reply:
Double-checked the generation procedures again and got you - multiple OAuth classes are available based on the used library, I'll change the JavaClientCodegen accordingly.